components/script/dom/dedicatedworkerglobalscope.rs

```diff
@@ -370,8 +371,10 @@ impl DedicatedWorkerGlobalScope {
                                     value);
 
         // Step 13.
-        let handled = !event.upcast::<Event>().fire(self.upcast::<EventTarget>());
-        if !handled {
+        let event_status = event.upcast::<Event>().fire(self.upcast::<EventTarget>());
+
+        // Step 15
+        if EventStatus::NotCanceled == event_status {
```

Comment:
If you want to swap around the left-hand and right-hand sides of these comparisons, I think that would read better. r+ either way, if you fix the other comment below.